common/cache_managers.py

```
@@ -0,0 +1,573 @@
+from loguru import logger
+
+from applications.API.models import InsultCategory
+from common.metrics import metrics
+
+
+class BaseCacheManager(ABC):
+    """
+    Abstract base class for cache managers with standardized interface.
+    """
+
+    def __init__(self, model_class: Type[models.Model], cache_prefix: str = None):
```

Ruff (0.12.2)

Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)